### PR TITLE
Add Support for Renaming of PythonMapInArrow [databricks]

### LINKED ISSUE #10673: [AUDIT] Rename plan nodes for PythonMapInArrowExec
**Describe the bug**

Spark Commit: https://github.com/apache/spark/commit/ed9a3a8b959322155b11a45650911c0e26a1a921 renamed plan nodes for Python 

`PythonMapInArrowExec/PythonMapInArrow` -> `MapInArrowExec/MapInArrow`

The RAPIDS Plugin code uses the `PythonMapInArrowExec` class from Spark. The shims for spark-4 should use the correct import

### PR DESCRIPTION
**Pull request by razajafri**
In Spark change https://github.com/apache/spark/commit/ed9a3a8b959322155b11a45650911c0e26a1a921, `PythonMapInArrow` has been changed to `MapInArrow`. This PR adds shims to support that change. 

fixes #10673

### CODE REVIEW

File: sql-plugin/src/main/spark400/scala/org/apache/spark/sql/rapids/shims/GpuMapInArrowExec.scala [hunk 1]
```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2022-2024, NVIDIA CORPORATION.
```
Comment (jlowe) [inline sql-plugin/src/main/spark400/scala/org/apache/spark/sql/rapids/shims/GpuMapInArrowExec.scala:2]:
```suggestion
 * Copyright (c) 2024, NVIDIA CORPORATION.
```

File: sql-plugin/src/main/spark400/scala/org/apache/spark/sql/rapids/shims/GpuMapInArrowExec.scala [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+import com.nvidia.spark.rapids._
+
+import org.apache.spark.api.python.PythonEvalType
+import org.apache.spark.sql.catalyst.expressions.{Attribute, Expression, PythonUDF}
+import org.apache.spark.sql.execution.SparkPlan
+import org.apache.spark.sql.execution.python.MapInArrowExec
+import org.apache.spark.sql.rapids.execution.python.GpuMapInBatchExec
+
+class GpuMapInArrowExecMetaBase(
```
Comment (jlowe) [inline sql-plugin/src/main/spark400/scala/org/apache/spark/sql/rapids/shims/GpuMapInArrowExec.scala:30]: I'm confused why this is here and not in GpuMapInArrowExecMeta, and also confusing why the base class has a convertToGpu that needs to be overridden by the only class that actually uses it.  Why have a base class if there's only one user, or why have a method that is not used in the base class?

Comment (razajafri) [reply sql-plugin/src/main/spark400/scala/org/apache/spark/sql/rapids/shims/GpuMapInArrowExec.scala]: I was fighting the compiler for potentially including a common base class, I gave up but then had some remnants left

**Comment by razajafri** [comment]
build

**Comment by razajafri** [comment]
build

**Comment by razajafri** [comment]
build

File: sql-plugin/src/main/spark400/scala/com/nvidia/spark/rapids/shims/MapInArrowExecShims.scala [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+  def execs: Map[Class[_ <: SparkPlan], ExecRule[_ <: SparkPlan]] = Seq(
+      GpuOverrides.exec[MapInArrowExec](
+        "The backend for Map Arrow Iterator UDF. Accelerates the data transfer between the" +
+          " Java process and the Python process. It also supports scheduling GPU resources" +
+          " for the Python process when enabled.",
+        ExecChecks((TypeSig.commonCudfTypes + TypeSig.ARRAY + TypeSig.STRUCT).nested(),
+          TypeSig.all),
+        (mapPy, conf, p, r) => new GpuMapInArrowExecMeta(mapPy, conf, p, r) {
+          override def tagPlanForGpu(): Unit = {
```
Comment (NVnavkumar) [inline sql-plugin/src/main/spark400/scala/com/nvidia/spark/rapids/shims/MapInArrowExecShims.scala:41]: Any specific reason this method is overridden here as opposed to `sql-plugin/src/main/spark400/scala/org/apache/spark/sql/rapids/shims/GpuMapInArrowExecMeta.scala`?

Comment (razajafri) [reply sql-plugin/src/main/spark400/scala/com/nvidia/spark/rapids/shims/MapInArrowExecShims.scala]: Thanks for the pointer. I have addressed it. PTAL

**Comment by razajafri** [comment]
build

**Review by jlowe — APPROVED** [review]

**Pull request merged**